### ISSUE TITLE
Eliminate prompt characters in examples from copyable text

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#267]
It would be nice if the prompt characters in examples were easily copyable and pasteable somehow. For example, if this showed up on the document:

```
>>> for i in xrange(32)
...     print i
```

And you selected it, only these bits were selected.

```
for i in xrange(32)
    print i
```

This would make it easier to actually paste the examples into an interpreter.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/267
- Originally reported by: Michael Hoffman
- Originally created at: 2009-10-06T17:05:39.855

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2009-11-08 17:24:12+00:00_

I assume this would require a piece of JavaScript that removes the prompts before copying.

The only question is what to do with output between prompts -- remove it as well?  Many doctest-style examples are meant to be executed one after the other in a shell; larger samples should not have doctest markup.

Anyway, let's keep this open as a nice-to-have.

**Comment by shimizukawa** [comment]
_From Gregg Lind on 2010-04-02 20:56:44+00:00_

After some discussion with Taggnoster, we wrote some code to handle this.

see:  1800:bfc9819c802f

**Comment by shimizukawa** [comment]
_From [tpowers](https://bitbucket.org/tpowers) on 2010-04-27 04:11:08+00:00_

When I need to do this sort of thing (rarely), the easiest thing to do is to fire up wxPython's pycrust or pyshell and just use the Paste or Paste Plus commands.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2010-08-05 17:46:33+00:00_

Issue #389 was marked as a duplicate of this issue.

**Comment by shimizukawa** [comment]
_From Ram Rachum on 2011-01-14 08:22:44+00:00_

Issue #604 was marked as a duplicate of this issue.

**Comment by shimizukawa** [comment]
_From Ram Rachum on 2011-01-14 08:25:47+00:00_

The annoying thing about having JS copy-paste trickery is that the user doesn't know that it's there. The user might avoid copy-pasting shell sessions completely after he sees that the prompts get selected together with the code, without even trying to paste so he won't see that the ">>> " parts disappear.

Do you think it's possible to automatically generate tables so that the prompts are in a separate cell than the code?

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-01-15 08:51:49+00:00_

Tables might work but for interleaved output, which needs to be skipped as well.

**Comment by stefan-balke** [comment]
Has this been addressed already?

**Comment by ankostis** [comment]
This issue is hard to find, and hence so many duplicates.
In particular, the title is missing the words
- clipboard
- copy
- doctest

Can someone with enough privileges amend the title?

**Comment by ankostis** [comment]
Could [this CSS trick is releveant](https://danoc.me/blog/css-prevent-copy//), relying on `data-pseudo-content--before`, work?

**Comment by tk0miya** [comment]
I use `user-select` style for this. see #6727 .

**Comment by tk0miya** [comment]
Fixed by #6727 .  (It's 10yrs after posted!)
Thanks,
